File: Semestrul 4/MPP2/Client-Server_2Limbaje/Problema8SC/Problema8SC/Problema8Model/src/main/java/problema8/model/Event.java
package problema8.model;

import java.util.Objects;

public class Event extends Entity<Long> {
    private String Name;
    private Integer Distance;
    private Long AgeGroupId;

    public Event(Long id, String name, Integer distance, Long ageGroupId) {
        setId(id);
        this.Name = name;
        this.Distance = distance;
        this.AgeGroupId = ageGroupId;
    }

    public String getName() {
        return Name;
    }

    public void setName(String name) {
        this.Name = name;
    }

    public Integer getDistance() {
        return Distance;
    }

    public void setDistance(Integer distance) {
        this.Distance = distance;
    }

    public Long getAgeGroupId() {
        return AgeGroupId;
    }

    public void setAgeGroupId(Long ageGroupId) {
        this.AgeGroupId = ageGroupId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Event event = (Event) o;
        return Objects.equals(Name, event.Name) && Objects.equals(Distance, event.Distance) && Objects.equals(AgeGroupId, event.AgeGroupId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(Name, Distance, AgeGroupId);
    }

    @Override
    public String toString() {
        return "Event{" +
                "name='" + Name + '\'' +
                ", distance=" + Distance +
                ", ageGroupId=" + AgeGroupId +
                '}';
    }
}

File: Semestrul 4/MPP2/Client-Server_2Limbaje/Problema8SC_CSharp/Model/Event.cs
namespace Problema8SC_CSharp.Model;

public class Event : Entity<long>
{
    public long Id
    {
        get => GetId();
        set => SetId(value);
    }
    public String Name { get; set; }
    public int Distance { get; set; }
    public long AgeGroupId { get; set; }
    
    public Event(long id, String name, int distance, long ageGroupId)
    {
        Id = id;
        Name = name;
        Distance = distance;
        AgeGroupId = ageGroupId;
    }
    
    public override String ToString()
    {
        return $"Name: {Name}, Distance: {Distance}, AgeGroupId: {AgeGroupId}";
    }
    
    public override bool Equals(object? obj)
    {
        if (obj == null || GetType() != obj.GetType())
        {
            return false;
        }
        
        Event e = (Event)obj;
        return Name.Equals(e.Name) && Distance == e.Distance && AgeGroupId == e.AgeGroupId;
    }

    protected bool Equals(Event other)
    {
        return Name == other.Name && Distance == other.Distance && AgeGroupId == other.AgeGroupId;
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(Name, Distance, AgeGroupId);
    }
}